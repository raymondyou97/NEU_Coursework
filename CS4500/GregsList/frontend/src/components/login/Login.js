'use es6';

import React, { PureComponent } from 'react';

import Button from 'react-bootstrap/Button';
import FormControl from 'react-bootstrap/FormControl';
import Form from 'react-bootstrap/Form';

import '../../styles/login.css';

class Login extends PureComponent {
  updateUsername = e => {
    if (this.props.error) this.props.setError(false);
    this.props.updateUsername(e.target.value);
  };

  updatePassword = e => {
    if (this.props.error) this.props.setError(false);
    this.props.updatePassword(e.target.value);
  };

  loginUser = () => {
    this.props.validateUser();
  };

  renderSubmit() {
    const { username, password } = this.props;

    const isButtonEnabled = username.length && password.length;

    return (
      <Button
        className="submit-button"
        disabled={!isButtonEnabled}
        onClick={this.loginUser}
        variant="primary"
      >
        Log in
      </Button>
    );
  }

  renderInputForms() {
    const { error } = this.props;

    return (
      <Form>
        <Form.Label className="login-label">Username</Form.Label>
        <FormControl
          className={`login-input ${error ? 'error' : null} username-input`}
          onChange={this.updateUsername}
          placeholder="Username"
          value={this.props.username}
        />
        <Form.Label className="login-label">Password</Form.Label>
        <FormControl
          className={`login-input ${error ? 'error' : null} password-input`}
          onChange={this.updatePassword}
          placeholder="Password"
          type="password"
          value={this.props.password}
        />
        {error ? (
          <Form.Label className="login-label-fail">
            Username and password do not match!
          </Form.Label>
        ) : null}
        {this.renderSubmit()}
      </Form>
    );
  }

  render() {
    return (
      <div className="sign-in-container">
        <h1>Sign in</h1>
        {this.renderInputForms()}
      </div>
    );
  }
}

export default Login;
