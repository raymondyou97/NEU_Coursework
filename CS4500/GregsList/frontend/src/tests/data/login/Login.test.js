'use es6';

import React from 'react';
import Login from '../../../components/login/Login';
import renderer from 'react-test-renderer';

test('Empty snapshot renders', () => {
  const component = renderer.create(
    <Login
      validateUser={() => {}}
      updateUsername={() => {}}
      updatePassword={() => {}}
      setError={() => {}}
      username={''}
      password={''}
      error={false}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Login with user data renders', () => {
  const component = renderer.create(
    <Login
      validateUser={() => {}}
      updateUsername={() => {}}
      updatePassword={() => {}}
      setError={() => {}}
      username={'jonah'}
      password={'password'}
      error={false}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Snapshot with data and error renders', () => {
  const component = renderer.create(
    <Login
      validateUser={() => {}}
      updateUsername={() => {}}
      updatePassword={() => {}}
      setError={() => {}}
      username={'asdf'}
      password={'password'}
      error={true}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
