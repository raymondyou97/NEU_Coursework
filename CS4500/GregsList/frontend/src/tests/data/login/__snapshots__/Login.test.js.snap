// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty snapshot renders 1`] = `
<div
  className="sign-in-container"
>
  <h1>
    Sign in
  </h1>
  <form
    className=""
  >
    <label
      className="login-label form-label"
    >
      Username
    </label>
    <input
      className="login-input null username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value=""
    />
    <label
      className="login-label form-label"
    >
      Password
    </label>
    <input
      className="login-input null password-input form-control"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <button
      className="submit-button btn btn-primary"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Register
    </button>
  </form>
</div>
`;

exports[`Login with user data renders 1`] = `
<div
  className="sign-in-container"
>
  <h1>
    Sign in
  </h1>
  <form
    className=""
  >
    <label
      className="login-label form-label"
    >
      Username
    </label>
    <input
      className="login-input null username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value="jonah"
    />
    <label
      className="login-label form-label"
    >
      Password
    </label>
    <input
      className="login-input null password-input form-control"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="password"
    />
    <button
      className="submit-button btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Register
    </button>
  </form>
</div>
`;

exports[`Snapshot with data and error renders 1`] = `
<div
  className="sign-in-container"
>
  <h1>
    Sign in
  </h1>
  <form
    className=""
  >
    <label
      className="login-label form-label"
    >
      Username
    </label>
    <input
      className="login-input error username-input form-control"
      onChange={[Function]}
      placeholder="Username"
      value="asdf"
    />
    <label
      className="login-label form-label"
    >
      Password
    </label>
    <input
      className="login-input error password-input form-control"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="password"
    />
    <label
      className="login-label-fail form-label"
    >
      Username and password do not match!
    </label>
    <button
      className="submit-button btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Register
    </button>
  </form>
</div>
`;
